packages/dev-container/src/electron-node/dev-container-file-service.ts

```diff
@@ -54,7 +49,7 @@ export class DevContainerFileService {
     }
 
     protected async searchForDevontainerJsonFiles(directory: string, depth: number): Promise<string[]> {
-        if (depth < 0) {
+        if (depth < 0 || !fs.existsSync(directory)) {
```

Comment:
Please don't use sync file system code in the backend - it blocks the whole server process and is bad practice.